contracts/CozyInvestCurve3CryptoEth.sol

```diff
@@ -0,0 +1,160 @@
+ * ETH money market and using that ETH to add liquidity to the Curve tricrypto USDT/WBTC/WETH pool
+ * @dev This contract is intended to be used by delegatecalling to it from a DSProxy
+ */
+contract CozyInvestCurve3CryptoEth {
+  /// @dev OpenZeppelin's Address and SafeERC20 implementations are used for addresses and ERC20 tokens
+  using Address for address payable;
+  using SafeERC20 for IERC20;
+
+  /// @notice Cozy protection market with ETH underlying to borrow from: Curve 3Crypto Trigger
+  address public immutable protectionMarket;
+
+  /// @notice Cozy money market with ETH underlying
+  address public immutable moneyMarket;
+
+  /// @notice Curve tricrypto Deposit Zap -- helper contract for wrapping ETH before depositing
+  ICrvDepositZap public immutable depositZap;
+
+  /// @notice Curve tricrypto Liquidity Gauge -- contract for measuring liquidity provided over time
+  /// and distributing reward tokens
+  ICrvGauge public immutable gauge;
+
+  /// @notice Curve tricrypto receipt token
+  IERC20 public immutable curveLpToken;
+
+  /// @notice Maximillion contract for repaying ETH debt
+  IMaximillion public immutable maximillion;
+
+  /// @dev Index of WETH in the curve `coins` mapping
+  uint256 internal constant ethIndex = 2;
+
+  constructor(
+    address _moneyMarket,
+    address _protectionMarket,
+    address _maximillion,
+    address _depositZap,
+    address _gauge
+  ) {
+    moneyMarket = _moneyMarket;
+    protectionMarket = _protectionMarket;
+    maximillion = IMaximillion(_maximillion);
+    gauge = ICrvGauge(_gauge);
+
+    depositZap = ICrvDepositZap(_depositZap);
+    curveLpToken = IERC20(depositZap.token());
+  }
+
+  /**
+   * @notice Protected invest method for borrowing from given cozy ETH market,
+   * and using that ETH to add liquidity to the Curve tricrypto pool
+   * @param _ethMarket Address of the market to borrow ETH from
+   * @param _borrowAmount Amount of ETH to borrow and deposit into Curve
+   * @param _curveMinAmountOut The minAmountOut we expect to receive when adding liquidity to Curve
+   */
+  function invest(
+    address _ethMarket,
+    uint256 _borrowAmount,
+    uint256 _curveMinAmountOut
+  ) external payable {
+    require(_ethMarket == moneyMarket || _ethMarket == protectionMarket, "Invalid borrow market");
+    ICozyEther _market = ICozyEther(_ethMarket);
+
+    // Borrow ETH from Cozy market
+    require(_market.borrow(_borrowAmount) == 0, "Borrow failed");
+
+    // Add liquidity to Curve, which returns a receipt token
+    uint256 _balance = depositZap.add_liquidity{value: _borrowAmount}(
+      [0, 0, _borrowAmount],
+      _curveMinAmountOut,
+      address(this)
+    );
+
+    // Approve the Curve tricrypto liquidity gauge to spend our receipt tokens using OpenZeppelin's
+    // SafeERC20 safeApprove function
+    curveLpToken.safeApprove(address(gauge), 0);
+    curveLpToken.safeApprove(address(gauge), type(uint256).max);
```

Comment:
In the cozy docs PR, I left some comments around adding comments to this section / other parts of the contract for clarity for reads. Let's make those same changes here so the docs and this guide are consistent

Reply:
same follow up comment here around using our own `TransferHelper` contract instead of OZ's safe approve